File: backend/app/schemas/review.py
from pydantic import BaseModel, model_validator
from typing import List, Dict, Optional


class ReviewRequest(BaseModel):
    """Request to start a code review"""
    repo_id: str
    files: Optional[List[str]] = None  # Specific files to review
    agents: Optional[List[str]] = None  # Specific agents to run


class ReviewFinding(BaseModel):
    """A single finding from the review"""
    id: str
    agent_name: str
    severity: str  # critical, high, medium, low, info
    category: str
    title: str
    description: str
    file_path: str
    start_line: int
    end_line: Optional[int] = None
    suggestion: str
    code_snippet: str

    @property
    def line(self) -> int:
        return self.start_line

    @model_validator(mode="after")
    def auto_fill_end_line(self) -> "ReviewFinding":
        if self.end_line is None:
            self.end_line = self.line
        return self


class ReviewResponse(BaseModel):
    """Complete review response"""
    review_id: str
    repo_id: str
    status: str
    total_findings: int
    severity_counts: Dict[str, int]
    findings: List[ReviewFinding]


class ReviewStatus(BaseModel):
    """Status of an ongoing review"""
    review_id: str
    status: str  # pending, processing, completed, failed
    progress: int  # 0-100
    message: Optional[str] = None
    error: Optional[str] = None

File: frontend/src/types/review.ts
export interface ReviewFinding {
  id: string
  agent_name: string
  severity: 'critical' | 'high' | 'medium' | 'low' | 'info'
  category: string
  title: string
  description: string
  file_path: string
  start_line: number
  end_line: number
  suggestion: string
  code_snippet: string
}

export interface ReviewResponse {
  review_id: string
  repo_id: string
  status: string
  total_findings: number
  severity_counts: {
    critical: number
    high: number
    medium: number
    low: number
    info: number
  }
  findings: ReviewFinding[]
}

export interface ReviewStatus {
  review_id: string
  status: 'pending' | 'processing' | 'completed' | 'failed'
  progress: number
  message?: string
  error?: string
}

export interface RepoInfo {
  repo_id: string
  name: string
  url: string
  files_count: number
  languages: string[]
}

export interface FileNode {
  name: string
  path: string
  type: 'file' | 'directory'
  children?: FileNode[]
}
